app/controllers/embeddable/embeddable_answers_controller.rb

```diff
@@ -11,11 +11,6 @@ def set_answer
   end
 
   def authorize_run_access
-    begin
-      authorize!(:access, @answer.run)
-    rescue
-      user_id_mismatch()
-      render(nothing: true, status: :unauthorized)
-    end
+    raise_error_if_not_authorized_run(@answer.run)
```

Comment:
In the code replaced this is calling `render(nothing: true, status: :unauthorized)` instead of the template that is ultimately rendered by    `raise_error_if_not_authorized_run`.  I'd guess this is because it is in response to a post request or json request. So if we return html it might cause an un handled error in the javascript code. Perhaps this can be handled better by checking the format of the request and returning different things based on this format.